src/pages/trade/redirect.tsx

```diff
@@ -0,0 +1,38 @@
+'use client';
+
+import { redirect } from 'next/navigation';
+import { ChainRegistryClient } from '@penumbra-labs/registry';
+import { envQueryFn } from '@/shared/api/env/env.ts';
+import { useQuery } from '@tanstack/react-query';
+
+const redirectSymbolsQueryFn = async () => {
+  const { PENUMBRA_CHAIN_ID } = await envQueryFn();
+  const chainRegistryClient = new ChainRegistryClient();
+  const registry = await chainRegistryClient.remote.get(PENUMBRA_CHAIN_ID);
+  const allAssets = registry
+    .getAllAssets()
+    .toSorted((a, b) => Number(b.priorityScore - a.priorityScore));
+  const baseAsset = allAssets[0]?.symbol;
+  const quoteAsset = allAssets[1]?.symbol;
+  if (!baseAsset || !quoteAsset) {
+    throw new Error('Could not find symbols in registry');
+  }
+
+  return { baseAsset, quoteAsset };
+};
+
+export const RedirectToPair = () => {
+  const { data, isLoading, error } = useQuery({
+    queryKey: ['redirectSymbols'],
+    retry: 1,
+    queryFn: redirectSymbolsQueryFn,
+  });
+
+  if (error) {
+    return <div className='text-red-600'>{String(error)}</div>;
+  } else if (isLoading || !data) {
+    return <div className='text-white'>Loading...</div>;
+  } else {
+    redirect(`/trade/${data.baseAsset}/${data.quoteAsset}`);
+  }
+};
```

Comment:
Was finding it a bit annoying that we had a hard-coded pair here. This causes issues when going back and forth between testnet/mainnet. When ahead and created a hook to pull the highest priority pair and show that as the default.

Reply:
Definitely a bug in the implementation. I should have been filtering out delegation tokens, fixed! Also, added some sample priority scores: https://github.com/prax-wallet/registry/pull/109.

Reply:
Nice, confirming resolved on my end:

![dex-explorer-getting-better](https://github.com/user-attachments/assets/e87c352a-aa9d-45e4-9e82-d341296e1931)

Looks like the styling is sub-optimal but we can handle that in time.

Reply:
Indeed, but we have an issue for that: https://github.com/penumbra-zone/dex-explorer/issues/107